doc/design/scheduled-tasks.md

```diff
@@ -19,41 +19,66 @@ We would like to move this `assetsSummary` calculation into a celery task, but w
 We need a way to run this task only once per batch of uploads.
 
 ## Proposal
-Instead of adding a new task to the queue whenever something needs to be done, we use a scheduled job that runs the task only when necessary.
+Instead of adding a new task to the queue to calculate `assetsSummary`, we use a scheduled job that runs the task only when necessary.
 
-Since `assetsSummary` calculation and dandiset metadata validation both need to be triggered in the same circumstances, we'll consider them together.
+Since `assetsSummary` only needs to be recalculated when an asset is added, removed, or modified, we don't want it to recompute whenever the Version is modified, as it is now.
 
 #### Request thread
-Whenever a Version is modified, a flag `needs_recomputing` is set on the Version indicating that it has been modified (instead of queueing a task as it does now):
+Whenever an asset is added (`POST`), removed (`DELETE`), or modified (`PUT`), a flag `needs_recomputing` is set on the Version indicating that it needs to be recomputed (instead of queueing a task as it does now):
 ```
-  # Things that modify the version, necessitating a validation/assetsSummary computation
-  ...
   # validate_version_metadata.delay(version.id)
   version.needs_recomputing = True
 ```
 
 #### Scheduled job
 Separately, a scheduled job that runs on a configurable fairly tight interval will check for any Versions that need to be updated.
-For each of those Versions, it also checks that the Version hasn't been modified in a certain amount of time (say 5 minutes). This prevents DB locks from happening while Assets are still being uploaded to the Version.
 
-The code would look something like this:
+The full code would look something like this:
 ```
 def scheduled_task():
-  versions = Version.objects.filter(needs_recomputing=True).filter(modified__lt=timezone.now() - 5 minutes)
-  # SELECT ... FROM api_version WHERE api_version.needs_recomputing AND now() - api_version.modified > 5 minutes;
+  with transaction.atomic():
```

Comment:
sorry , but can't grasp from the [doc](https://docs.djangoproject.com/en/3.2/topics/db/transactions/#controlling-transactions-explicitly) -- is `.atomic` blocking the entire DB from any writes or just not performing `DB writes` of changes until it is done?

if the former (total DB block) -- wouldn't we screw up interaction from other users/tasks with the DB?

if the later (no writes until out) -- wouldn't `version.needs_recomputing = False` and `version.status = PENDING` have no effect for other "tasks" which would check on them?

Reply:
`atomic` is just a regular DB transaction.
`needs_recomputing` will remain `True` until the task commits the transaction successfully, so the Version cannot be published. This is the only place where `needs_recomputing` is set to `False`, so the Version is considered invalid for publishing until the task succeeds.

If the transaction were to fail, the next scheduled job would pick it up as if the first job had never run. Eventually the transaction should succeed.